api/agentic/v1alpha1/backend_types.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// IMPORTANT: Run "make generate-all" to regenerate code after modifying this file
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BackendSpec defines the desired state of Backend.
+type BackendSpec struct {
+	// Type specifies the type of the backend.
+	// Currently, only "MCP" is supported.
+	// +required
+	Type *BackendType `json:"type"`
+
+	// MCP defines a MCP backend.
+	// +optional
+	MCP MCPBackend `json:"mcp,omitempty"`
+}
+
+// BackendType defines the type of the Backend.
+// +kubebuilder:validation:Enum=MCP
+type BackendType string
+
+const (
+	// BackendTypeMCP defines the type of the backend as MCP.
+	BackendTypeMCP BackendType = "MCP"
+)
+
+// MCPBackend describes a MCP Backend.
+// ServiceName and Hostname cannot be defined at the same time.
+// +kubebuilder:validation:ExactlyOneOf=serviceName;hostname
+type MCPBackend struct {
+	// ServiceName defines the Kubernetes Service name of a MCP backend.
+	// +optional
+	ServiceName string `json:"serviceName,omitempty"`
+
+	// Hostname defines the hostname of the external MCP service to connect to.
+	// +optional
+	Hostname string `json:"hostname,omitempty"`
```

Comment:
FYI, https://github.com/kubernetes-sigs/wg-ai-gateway/pull/20/files for egress Backends, with some MCP provisions

Reply:
Okay thanks for the pointer to the context @LiorLieberman ! Thanks for flagging this @david-martin.
I think we're good now? cc @haiyanmeng

Reply:
@david-martin just making sure this was the intention? Or did you mean something else?

Reply:
Yes, there are different fields. Should we consider them here?
OK if not, to focus on a specific use case here first for prototyping.

Also, mentioned that PR for general awareness as there'll likely be some convergence in future.

Reply:
I would like to align with what Keith is working on in the short term. Can we have an issue to track this? I dont want to take anything as https://github.com/kubernetes-sigs/wg-ai-gateway/pull/20 is still being iterated I believe.